### PR TITLE
Redesigned background task to be long-running and added logger

### PR DESCRIPTION
**Pull request by amir734jj**
- It's very important to use `TaskCreationOptions.LongRunning` if the task is long-running. This will schedule garbage collector differently and optimize memory allocation for long-running.
- `CancellationToken` to signal task to stop
- Added console logger as well.

### CODE REVIEW

File: Deployr.Processor/BackgroundDeployrTask.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    public class BackgroundDeployrTask : IBackgroundDeployrTask
+    {
+        /// <summary>
+        /// Wait period in Millisecond
+        /// </summary>
+        private const int WaitMillisecond = 5000;
+
+        public CancellationTokenSource CancellationToken { get; }
```
Comment (TahaNKhan) [inline Deployr.Processor/BackgroundDeployrTask.cs]: CancellationTokenSource

File: Deployr.Processor/BackgroundDeployrTask.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+                    logger.LogTrace("Finished running listener task");
+
+                    logger.LogTrace($"Starting to wait {WaitMillisecond}ms before the next run");
+                    
+                    Thread.Sleep(WaitMillisecond);
+
+                    logger.LogTrace($"Finished Waiting {WaitMillisecond}ms for the next run");
+                }
+            }, null, token, TaskCreationOptions.LongRunning, TaskScheduler.Default);
```
Comment (TahaNKhan) [inline Deployr.Processor/BackgroundDeployrTask.cs:58]: Good call

File: Deployr.Processor/BackgroundDeployrTask.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+        public BackgroundDeployrTask(IProcessorTask processorTask, ILogger<BackgroundDeployrTask> logger)
+        {
+            CancellationToken = new CancellationTokenSource();
+
+            var token = CancellationToken.Token;
+
+            ListenerTask = Task.Factory.StartNew(_ =>
+            {
+                while (!CancellationToken.IsCancellationRequested)
```
Comment (TahaNKhan) [inline Deployr.Processor/BackgroundDeployrTask.cs]: All of this should be surrounded by a try catch block, so we don’t exit the application on an exception

**Review by TahaNKhan — CHANGES_REQUESTED** [review]

**Comment by amir734jj** [comment]
Updated the code.

**Review by TahaNKhan — APPROVED** [review]

**Pull request merged**